src/main/java/ben/tasks/TaskList.java

```diff
@@ -0,0 +1,201 @@
+                    String endDate = tokens[4];
+
+                    try {
+                        LocalDate dateFormattedStartDate = LocalDate.parse(startDate);
+                        LocalDate dateFormattedEndDate = LocalDate.parse(endDate);
+                        this.tasks.add(new Event(isDone, description, dateFormattedStartDate, dateFormattedEndDate));
+                        break;
+                    } catch (DateTimeParseException e) {
+                        throw new BenException("Invalid deadline format");
+                    }
+
+                default:
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Checks if the TaskList is empty.
+     *
+     * @return true if the TaskList is empty, false otherwise.
+     */
+    public boolean isEmpty() {
+        return this.tasks.isEmpty();
+    }
+
+    /**
+     * Checks if the given index is within the bounds of the TaskList.
+     *
+     * @param index The index to check.
+     * @return true if the index is within bounds, false otherwise.
+     */
+    public boolean isOutOfBounds(int index) {
+        return index < 0 || index >= tasks.size();
+    }
+
+    /**
+     * Gets the number of tasks in the TaskList.
+     *
+     * @return The number of tasks in the TaskList.
+     */
+    public int size() {
+        return this.tasks.size();
+    }
+
+    /**
+     * Marks a task at the specified index as done.
+     *
+     * @param index The index of the task to mark as done.
+     */
+    public void markTask(int index) {
+        Task currTask = this.tasks.get(index);
+        currTask.markTask();
+    }
+
+    /**
+     * Unmarks a task at the specified index as not done.
+     *
+     * @param index The index of the task to unmark.
+     */
+    public void unmarkTask(int index) {
+        Task currTask = this.tasks.get(index);
+        currTask.unmarkTask();
+    }
+
+    /**
+     * Gets the string representation of a task at the specified index.
+     *
+     * @param index The index of the task.
+     * @return The string representation of the task.
+     */
+    public String toString(int index) {
+        Task currTask = this.tasks.get(index);
+        return currTask.toString();
+    }
+
+    /**
+     * Displays the entire TaskList.
+     */
+    public void showTaskList() {
+        for (int i = 0; i < this.tasks.size(); i++) {
+            Task currTask = this.tasks.get(i);
+            Ui.show((i + 1) + ". " + currTask);
+        }
+    }
+
+    /**
+     * Adds a task to the TaskList.
+     *
+     * @param task The task to be added.
+     */
+    public void addTask(Task task) {
+        this.tasks.add(task);
+    }
+
+    /**
+     * Removes a task at the specified index from the TaskList.
+     *
+     * @param index The index of the task to be removed.
+     * @return The removed task.
+     */
+    public Task removeTask(int index) {
+        Task deletedTask = this.tasks.remove(index);
+        return deletedTask;
+    }
```

Comment:
I like that your methods (elsewhere as well) are named as verbs and written in camelCase, following the coding standards